drivers/opamp/opamp_stm32g4x_opamp.c

```diff
@@ -0,0 +1,666 @@
+	struct stm32_opamp_data *data = dev->data;
+	const struct adc_dt_spec *adc_ch = &cfg->adc_ch;
+	OPAMP_TypeDef *opamp = cfg->opamp;
+	int ret = 0;
+
+	ret = device_is_ready(clk);
+	if (!ret) {
+		LOG_ERR("%s clock control device not ready (%d)", dev->name, ret);
+		return ret;
+	}
+
+	if (adc_ch->dev != NULL) {
+		if (!adc_is_ready_dt(adc_ch)) {
+			LOG_ERR("%s ADC device not ready", adc_ch->dev->name);
+			return -ENODEV;
+		}
+	}
+
+	/* Enable OPAMP bus clock */
+	ret = clock_control_on(clk, (clock_control_subsys_t)&cfg->pclken[0]);
+	if (ret != 0) {
+		LOG_ERR("%s clock op failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	/* Enable OPAMP clock source if provided */
+	if (cfg->pclk_len > 1) {
+		ret = clock_control_configure(clk, (clock_control_subsys_t)&cfg->pclken[1], NULL);
+		if (ret != 0) {
+			LOG_ERR("%s clock configure failed (%d)", dev->name, ret);
+			return ret;
+		}
+	}
+
+	/* Configure OPAMP inputs as specified in Device Tree, if any */
+	ret = pinctrl_apply_state(cfg->pincfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0 && ret != -ENOENT) {
+		/*
+		 * If the OPAMP is used only with internal channels, then no pinctrl is
+		 * provided in Device Tree, and pinctrl_apply_state returns -ENOENT,
+		 * but this should not be treated as an error.
+		 */
+		LOG_ERR("%s pinctrl setup failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	k_mutex_lock(&data->dev_mtx, K_FOREVER);
+
+	/* Power mode shall be set before calibration - since calibration is mode dependent */
+	LL_OPAMP_SetPowerMode(opamp, cfg->power_mode);
+
+	LL_OPAMP_SetInternalOutput(opamp, cfg->adc_ch.dev != NULL
+						  ? LL_OPAMP_INTERNAL_OUTPUT_ENABLED
+						  : LL_OPAMP_INTERNAL_OUTPUT_DISABLED);
+
+	ret = stm32_opamp_self_calibration(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* Apply trimming values defined explicitly in DTS */
+	if (cfg->pmos_trimming_value <= 0x1f || cfg->nmos_trimming_value <= 0x1f) {
+		/* Enable user trimming mode if one of trimming values is given
+		 * explicitly in DTS.
+		 */
+		LL_OPAMP_SetTrimmingMode(opamp, LL_OPAMP_TRIMMING_USER);
+
+		if (cfg->pmos_trimming_value <= 0x1f) {
+			LL_OPAMP_SetTrimmingValue(opamp, LL_OPAMP_TRIMMING_PMOS,
+						  cfg->pmos_trimming_value);
+		}
+		if (cfg->nmos_trimming_value <= 0x1f) {
+			LL_OPAMP_SetTrimmingValue(opamp, LL_OPAMP_TRIMMING_NMOS,
+						  cfg->nmos_trimming_value);
+		}
+	}
+
+	ret = stm32_opamp_set_functional_mode(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	LL_OPAMP_SetInputsMuxMode(opamp, cfg->inputs_mux_mode);
+
+	k_mutex_unlock(&data->dev_mtx);
+	return pm_device_driver_init(dev, stm32_opamp_pm_callback);
+}
+
+#define STM32_OPAMP_DT_POWER_MODE(inst)                                                            \
+	CONCAT(LL_OPAMP_POWERMODE_, DT_INST_STRING_TOKEN(inst, st_power_mode))
+
+#define STM32_OPAMP_DT_INPUTS_MUX_MODE(inst)                                                       \
+	CONCAT(LL_OPAMP_INPUT_MUX_, DT_INST_STRING_TOKEN_OR(inst, st_inputs_mux_mode, DISABLE))
```

Comment:
```suggestion
	CONCAT(LL_OPAMP_INPUT_MUX_, DT_INST_STRING_TOKEN(inst, st_inputs_mux_mode))
```
`DISABLED` is already the default value for this property.